cmake/modules/AddSwift.cmake

```diff
@@ -131,7 +131,12 @@ function(_add_host_variant_c_compile_flags target)
 
   is_build_type_optimized("${CMAKE_BUILD_TYPE}" optimized)
   if(optimized)
-    target_compile_options(${target} PRIVATE -O2)
+    is_build_type_optimized_for_size("${CMAKE_BUILD_TYPE}" optimized_for_size)
+    if(optimized_for_size)
+      target_compile_options(${target} PRIVATE -Os)
+    else()
+      target_compile_options(${target} PRIVATE -O2)
```

Comment:
Another thing to look at is from LLVM IIRC there is a command line flag -O that we import. Can you just validate that this overrides that? (I believe this comes later on the command line so it overrides, but it would be good to check).

Reply:
Yes, I checked, this is present later on the command lines.

Reply:
Ok good. Thanks! I just wanted to be careful it didn't change.

Reply:
Wait, I'm very confused why these are needed at all.  You are checking `CMAKE_BUILD_TYPE` - the flags for that should already be set properly (though it may do `-O3` instead of `-O2`).  What did I overlook here?  `CMAKE_<LANG>_FLAGS_${CMAKE_BUILD_TYPE}` is the right variable to use rather than hardcoding `-Os` and `-O2` - and it will ensure that things are handled properly for MSVC as well.